specs/electra/beacon-chain.md

```diff
@@ -280,7 +279,7 @@ class AttesterSlashing(Container):
 class Attestation(Container):
     aggregation_bits: Bitlist[MAX_VALIDATORS_PER_COMMITTEE * MAX_COMMITTEES_PER_SLOT]  # [Modified in Electra:EIP7549]
     data: AttestationData
-    committee_bits: Bitvector[MAX_COMMITTEES_PER_SLOT]  # [New in Electra:EIP7549]
+    committee_indices: List[CommitteeIndex, MAX_COMMITTEES_PER_SLOT]  # [New in Electra:EIP7549]
```

Comment:
I wonder if we should enforce these committee indices to be sorted

Reply:
tbh, I don’t know, do we have any reason for enforcing them to be sorted?

Reply:
Nothing other than it being cheap to do and nice to display, but whoever displays it can also sort it out of band. I'm indifferent, but thought it's worth considering so I don't miss anything.

Reply:
Yeah, it worth considering! but beyond potential UX improvement I don’t see any reason to enforce that. So by default i’d leave it as is unless there are other reasons for doing that

Reply:
I feel like enforcing sorting makes things more clear and preserve equality: with non-sorted we may have `Attestation1.hash_root` != `Attestation2.hash_root` but they could have exact same coverage. (correct me if i'm wrong)
Having a unique way of expressing the same thing maybe isn't super important but seems a nice property to have.

Reply:
This is a great point! The enforcement makes a lot of sense from this perspective

Reply:
as per offline discussion, worth noting that, as we introduce potential duplications in index list, then the the assumption `same attestation -> same hash_root` will be broken anyway

Reply:
> as we introduce potential duplications in index list

duplications would imply that signature is different already (since each entry represents a term being added to the aggregate sig), which was the motivation for this PR..

I'd be interested to see how much of an improvement this possibility actually provides, ie how often is it that when we're packing attestations, we would benefit from overlapping aggregates - this would imply that aggregators observed different attestations which _should_ be rare.

Reply:
If we take a look at on chain data (i picked a few random blocks — not that representative tho), they have duplicates. Probably, those duplicates are added by the proposer from what it has observed in the attestation subnet itself as the duplicates are usually scarce.

The issue that we have discussed with @tbenr is that the same attestation can have different representations. Consider we have `n` aggregates from the same committee in one on chain aggregate, if we enforce the order of committee indices in the array then by changing the order of the aggregation bits we can get up to `n!` different on chain aggregates with distinct `hash_tree_root()` each that are actually carrying the same information.

Reply:
> changing the order of the aggregation bits 

Sure, as long as they're permutations - I misunderstood it as entries with different number of duplicates being discussed.

I'm bringing it up because one alternative encoding is to list the validators that were _not_ included - usually, there's only a few of those and this way we would probably have a significantly smaller list, but that again brings us back to the only-one-item territory - it's been discussed as an optimisation strategy as well, ie keep an aggregated committee key and subtract - probably makes more sense for sync committee messages but still.